internal/stats/metrics_recorder_list.go

```diff
@@ -49,46 +48,46 @@ func NewMetricsRecorderList(shs []stats.Handler) *MetricsRecorderList {
 	}
 }
 
-func verifyLabels(labels []string, optionalLabels []string, labelsRecv ...string) {
-	if got, want := len(labelsRecv), len(labels)+len(optionalLabels); got != want {
-		logger.Fatalf("Length of labels passed to Record incorrect, got: %v, want: %v.", got, want)
+func verifyLabels(desc *estats.MetricDescriptor, labelsRecv ...string) {
+	if len(labelsRecv) != len(desc.Labels)+len(desc.OptionalLabels) {
+		panic(fmt.Sprintf("Length of labels passed to Record incorrect for metric %q.", desc.Name))
 	}
 }
 
 func (l *MetricsRecorderList) RecordInt64Count(handle *estats.Int64CountHandle, incr int64, labels ...string) {
-	verifyLabels(handle.Labels, handle.OptionalLabels, labels...)
+	verifyLabels((*estats.MetricDescriptor)(handle), labels...)
```

Comment:
Oh..I was thinking we wouldn't need to do the cast here for some reason...  I wonder if it's worth adding a method to the handles like:

```go
func (h handle) Descriptor() *estats.MetricDescriptor {
	return (*estats.MetricDescriptor)(h)
}
```

...in order to keep the ugliness out of the places we do the cast.

Reply:
Yeah, these are also cast like this in OTel, we could add but then that populates exported namespace. Although if a user does use the metrics registry and writes their own stats handlers, they will need to key on a property of the descriptor, and perhaps they will use the pointer as well.

Reply:
I'll keep it as a separate commit and you can see how it looks.

Reply:
Sent out a separate commit: https://github.com/grpc/grpc-go/pull/7428/commits/619b20e3640e89ca427550dd3b90410dc4d541c6, let me know what you think.